### PR TITLE
doc: Convert contributing/* to markdown

### LINKED ISSUE #120011: Convert contributing section from Docbook to CommonMark
This issue tracks the conversion of `/doc/contributing/*` from Docbook to CommonMark per [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).

If you intend to work on this conversion, please comment below!

## Conversion instructions

1. Comment here that you intend to work on it. (Please also post if you stop working on it.)
1. Checkout a new branch of nixpkgs master.
1. Rename `doc/contributing/*.xml` to `*.section.md`.
1. Convert the contents from Docbook to CommonMark. Follow [conversion best practices](https://github.com/NixOS/nixpkgs/issues/105243). Use existing .md sections for inspiration.
1. Use `nix-build` from the `doc/` folder to build the manual.
1. Preview the manual on your computer at http://0.0.0.0:8000/manual.html after running from doc/ folder:

    `nix-shell -p python3 --run 'python3 -m http.server --directory ./result/share/doc/nixpkgs'`

1. When it looks good, submit a PR and reference this issue.

### PR DESCRIPTION
**Pull request by bobby285271**
###### Motivation for this change

Closes #120011

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Converted xml to md and updated manual.xml to reference it.

### CODE REVIEW

File: doc/contributing/quick-start.chapter.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+To add a package to Nixpkgs:
+
+1. Checkout the Nixpkgs source tree:
+
+   ```ShellSession
+   $ git clone https://github.com/NixOS/nixpkgs
+   $ cd nixpkgs
+   ```
+
+2. Find a good place in the Nixpkgs tree to add the Nix expression for your package. For instance, a library package typically goes into `pkgs/development/libraries/pkgname`, while a web browser goes into `pkgs/applications/networking/browsers/pkgname`. See <xref linkend="sec-organisation" /> for some hints on the tree organisation. Create a directory for your package, e.g.
```
Comment (Mic92) [inline doc/contributing/quick-start.chapter.md:12]: Is this xref supposed to be here?

Comment (bobby285271) [reply doc/contributing/quick-start.chapter.md:12]: I guess this is equivalent to `See [Section 18.3, “File naming and organisation”](#sec-organisation) for some hints...`, by using `xref` we can always get the right section title, other files like [`multiple-output.chapter.md`](https://raw.githubusercontent.com/NixOS/nixpkgs/master/doc/stdenv/multiple-output.chapter.md) and [`stdenv.chapter.md`](https://raw.githubusercontent.com/NixOS/nixpkgs/master/doc/stdenv/stdenv.chapter.md) also use `xref`.

**Comment by fricklerhandwerk** [comment]
I also started dabbling at it a few days ago, and wondered how we can efficiently check for equivalency of the conversion. Just `diff`ing the resulting HTML is impractical, because potentially different formatting will produce a lot of noise, and the document is huge. It sure did on the `xml` intermediates. @bobby285271 did you try anything like that?

Side note: I could not build the `master` docs on macOS 10.14, because `/doc/doc-support/default.nix` does not properly build, which is weird: The build *succeeds*, but its `result` symlink goes nowhere, the target file in `/nix/store/` does not come into existence. It works on `release-20.09` though.

**Comment by fricklerhandwerk** [comment]
@Mic92 I understand from the fact that you just merged this that we don't care much? I would have liked to review this commit by running some kind of check, but I probably should have said so explicitly...

**Comment by Mic92** [comment]
> @Mic92 I understand from the fact that you just merged this that we don't care much? I would have liked to review this commit by running some kind of check, but I probably should have said so explicitly...

I checked it manually. Don't know how to efficiently automate this process. I think most of the conversion is already done. So I don't think adding checks now would a lot of value.

**Comment by fricklerhandwerk** [comment]
I didn't want to add persistent checks to the build, just make sure the new result is close enough to the old one. I'm afraid I'd break something unwittingly and wouldn't even notice, because it's tons of text to go through. But if looking over it is good enough for you, it's sure good enough for me.

**Comment by blaggacao** [comment]
Thank you @bobby285271 ! :heart:

A postponed ["docsgiving"](https://discourse.nixos.org/t/happy-docsgiving/10214?u=blaggacao) is just as much as joy.

**Pull request merged**